src/js/jsx/shared/Dialog.jsx

```diff
@@ -124,11 +132,10 @@ define(function (require, exports, module) {
                 id = this.props.id;
 
             if (this.state.open) {
+                this._target = null;
                 flux.actions.dialog.closeDialog(id);
             } else if (!this.props.disabled) {
-                this.setState({
-                    target: event.target
-                });
+                this._target = event.target;
```

Comment:
The question I have: does that happen before or after the render? Before or after componentDidUpdate? Before or after componentWillUpdate? What if the component doesn't update because of shouldComponentUpdate? So many questions!